packages/main/src/TabContainer.js

```diff
@@ -420,28 +419,113 @@ class TabContainer extends UI5Element {
 		ResizeHandler.deregister(this._getHeader(), this._handleResize);
 	}
 
-	_onTabStripClick(event) {
+	_setItemsExternalProperties(items) {
+		items.filter(item => !item.isSeparator).forEach((item, index, arr) => {
+			item._isInline = this.tabLayout === TabLayout.Inline;
+			item._mixedMode = this.mixedMode;
+			item._posinset = index + 1;
+			item._setsize = arr.length;
+			item._getTabContainerHeaderItemCallback = _ => this.getDomRef().querySelector(`#${item._id}`);
+			item._itemSelectCallback = this._onItemSelect.bind(this);
+			item._getRealDomRef = () => this.getDomRef().querySelector(`*[data-ui5-stable=${item.stableDomRef}]`);
+			item._realTab = this._selectedTab;
+			item._isTopLevelTab = this.items.some(i => i === item);
+			walk(items, tab => {
+				tab._realTab = item._realTab;
+			});
+		});
+	}
+
+	async _onTabStripClick(event) {
 		const tab = getTab(event.target);
 		if (!tab) {
 			return;
 		}
+		event.preventDefault();
+		event.stopPropagation();
+		if (event.target.hasAttribute("ui5-button")) {
+			this._onTabExpandButtonClick(event);
+			return;
+		}
 
+		if (!tab._realTab._hasOwnContent && tab._realTab.tabs.length) {
+			this._overflowItems = [];
+			this._overflowItems = tab._realTab.subTabs;
+			this._addStyleIndent(this._overflowItems, false);
+
+			this.responsivePopover = await this._respPopover();
+			if (this.responsivePopover.opened) {
+				this.responsivePopover.close();
+			} else {
+				this.responsivePopover.showAt(tab);
```

Comment:
The initial focus need to be set like in the _onTabExpandButtonClick function. The logic can be put in a separate function to avoid code duplication.